drivers/adc/ad7616/iio_ad7616.c

```diff
@@ -0,0 +1,293 @@
+/***************************************************************************//**
+* @file   iio_ad7616.c
+* @brief  Implementation of AD7616 IIO interface
+********************************************************************************
+* Copyright (c) 2023 Analog Devices, Inc.
+* Copyright (c) 2023 BayLibre, SAS.
+* All rights reserved.
+*
+* This software is proprietary to Analog Devices, Inc. and its licensors.
+* By using this software you agree to the terms of the associated
+* Analog Devices Software License Agreement.
+*******************************************************************************/
+#ifdef IIO_SUPPORT
+
+#include <stdio.h>
+#include <errno.h>
+
+#include "ad7616.h"
+#include "iio_ad7616.h"
+#include "iio.h"
+#include "no_os_util.h"
+
+#define AD7616_CH(_idx) {\
+	.name = "voltage" #_idx, \
+	.ch_type = IIO_VOLTAGE,\
+	.ch_out = IIO_DIRECTION_INPUT,\
+	.indexed = true,\
+	.channel = _idx,\
+	.scan_index = _idx,\
+	.scan_type = &ad7616_iio_scan_type,\
+	.attributes = ad7616_iio_ch_attributes\
+}
+
+#define AD7616_CHN_ATTR(_name, _priv) {\
+	.name = _name,\
+	.priv = _priv,\
+	.show = iio_ad7616_attr_get,\
+	.store = iio_ad7616_attr_set\
+}
+
+#define CHN_STORAGE_BITS 16
+
+struct ad7616_iio {
+	struct ad7616_dev *drv_dev;
+	struct iio_device *iio_dev;
+};
+
+enum ad7616_iio_attributes_id {
+	CRC_ATTR_ID,
+};
+
+static int iio_ad7616_attr_get(void *device, char *buf, uint32_t len,
+			       const struct iio_ch_info *channel,
+			       intptr_t priv)
+{
+	struct ad7616_dev *drv_dev = device;
+
+	switch (priv) {
+	case CRC_ATTR_ID:
+		return sprintf(buf, "%u", drv_dev->crc);
+	default:
+		return -1;
+	}
+}
+
+static int iio_ad7616_attr_set(void *device, char *buf, uint32_t len,
+			       const struct iio_ch_info *channel,
+			       intptr_t priv)
+{
+	switch (priv) {
+	case CRC_ATTR_ID:
```

Comment:
so looks like this is a draft, could you please show an output of iio_info, to see how this device looks from libiio point of view ?

Reply:
I've put a dump of `iio_info` in the PR description